packages/react-instantsearch-hooks-dom/src/ui/__tests__/Menu.test.tsx

```diff
@@ -0,0 +1,263 @@
+import { render } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import React from 'react';
+
+import { Menu } from '../Menu';
+
+import type { MenuProps } from '../Menu';
+
+describe('Menu', () => {
+  function createProps(props: Partial<MenuProps>): MenuProps {
+    let isShowingMore = false;
+    return {
+      items: [
+        {
+          value: 'Insignia™',
+          label: 'Insignia™',
+          count: 746,
+          isRefined: true,
+        },
+        {
+          value: 'Samsung',
+          label: 'Samsung',
+          count: 633,
+          isRefined: false,
+        },
+      ],
+      onRefine: jest.fn(),
+      createURL: (value) => `#${value}`,
+      canToggleShowMore: true,
+      onToggleShowMore: () => {
+        isShowingMore = !isShowingMore;
+      },
```

Comment:
Then you can add a test where you click the button and assert:
- That the function was called
- That the UI updated accordingly

```suggestion
      onToggleShowMore: jest.fn(() => {
        isShowingMore = !isShowingMore;
      }),
```

Reply:
added a test for just the function being called, because the state really is the responsibility of a wrapping component, so it doesn't need to be tested in detail here